src/mito2/src/memtable/merge_tree/data.rs

```diff
@@ -519,26 +522,137 @@ impl<'a> DataPartEncoder<'a> {
             true,
         )?;
         writer.write(&rb).context(error::EncodeMemtableSnafu)?;
-        let _file_meta = writer.close().context(error::EncodeMemtableSnafu)?;
-        Ok(Bytes::from(bytes))
+        let metadata = writer.close().context(error::EncodeMemtableSnafu)?;
+        Ok(DataPart::Parquet(ParquetPart {
+            data: Bytes::from(bytes),
+            metadata,
+        }))
     }
 }
 
+/// Data parts under a shard.
+pub struct DataParts {
+    parts: Vec<DataPart>,
+}
+
 /// Format of immutable data part.
 pub enum DataPart {
-    Parquet(Bytes),
+    Parquet(ParquetPart),
 }
 
 impl DataPart {
     fn is_empty(&self) -> bool {
         match self {
-            DataPart::Parquet(data) => data.is_empty(),
+            DataPart::Parquet(p) => p.data.is_empty(),
+        }
+    }
+
+    /// Reads frozen data part and yields [DataBatch]es.
+    pub fn read(&self, _pk_weights: &[u16]) -> Result<DataPartReader> {
```

Comment:
We can remove `pk_weights` if it is unused.

Reply:
> We can remove `pk_weights` if it is unused.

fixed in ac863a7a33